Documentation/ceph-object-bucket-claim.md

```diff
@@ -102,18 +102,94 @@ metadata:
   labels:
     aws-s3/object [1]
 provisioner: rook-ceph.ceph.rook.io/bucket [2]
-parameters: [3]
+parameters:
   objectStoreName: my-store
   objectStoreNamespace: rook-ceph
-  region: us-west-1
+  region: us-west-1 [3]
   bucketName: ceph-bucket [4]
 reclaimPolicy: Delete [5]
 ```
 1. `label`(optional) here associates this `StorageClass` to a specific provisioner.
 1. `provisioner` responsible for handling `OBCs` referencing this `StorageClass`.
-1. **all** `parameter` required.
+1. `region`(optional) defines the region in which the bucket should be created. For RGW, this maps to [zonegroup](https://docs.ceph.com/en/latest/radosgw/multisite/#zone-groups) of the server. If user does not define `region`, Rook will fill this value accordingly. For examples please check at the end of the doc.
 1. `bucketName` is required for access to existing buckets but is omitted when provisioning new buckets.
 Unlike greenfield provisioning, the brownfield bucket name appears in the `StorageClass`, not the `OBC`.
 1. rook-ceph provisioner decides how to treat the `reclaimPolicy` when an `OBC` is deleted for the bucket. See explanation as [specified in Kubernetes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#retain)
 + _Delete_ = physically delete the bucket.
 + _Retain_ = do not physically delete the bucket.
+
+
+##### Example: Default region
+In the below example `region` is not defined, so Rook Operator will fetch `zonegroup` from the `RGW` server and fill this out.
```

Comment:
Can you add what the region is in this example? my-store?
```suggestion
In the below example `region` is not defined, so Rook Operator will default to the name of the `zonegroup`, which is the same as the object store name `my-store`. 
```

Reply:
@travisn: This is not completely correct, zonegroup in object context can have the following values.
1.) normal simple cluster -> name of the object-store
2.) external cluster -> `default` 
3.) if multisite configure then it will be value created by the user
Please check `getMultisiteForObjectStore()` for more details. In any of the cases, if the user does not define region value, the Rook operator will fetch the value from `object Store Context` and set it accordingly. Please note if the user provides custom `region` then he configure multisite settings but if RGW configured with multisite it does not mean user need to provide the `region` value

Reply:
> @travisn: This is not completely correct, zonegroup in object context can have the following values. 
> 1.) normal simple cluster -> name of the object-store 

Ok, this is what I expect this first example to be... the local simple cluster. 

> 2.) external cluster -> `default` 

Perhaps we need an external cluster example that points out it could be anything, depending on the external cluster config.

> 3.) if multisite configure then it will be value created by the user 
> Please check `getMultisiteForObjectStore()` for more details. In any of the cases, if the user does not define region value, the Rook operator will fetch the value from `object Store Context` and set it accordingly. Please note if the user provides custom `region` then he configure multisite settings but if RGW configured with multisite it does not mean user need to provide the `region` value

This case I thought was the second example below. If it's not just the zonegroup name, then it's very confusing to me. We need to figure out how to simplify the explanation in the docs, we can't ask the users to look at the code.

Reply:
Could we rewrite the new documentation for `region` and the examples to be very clear on these two points? I'm still not clear on these:
1. us-east-1 will always work for their region 
2. They can use a different region by following the docs in this PR, but it is totally optional

Reply:
> Could we rewrite the new documentation for `region` and the examples to be very clear on these two points? I'm still not clear on these:
> 
>     1. us-east-1 will always work for their region
> 
@travisn I have mentioned it in [second example](https://github.com/rook/rook/pull/9754/files#diff-fb33f0fa41ca6346e7562087d2da05cc32fb65ea2587b23bf36b866a9a609720R137) and thought of keeping the first example leaving the `region` value. So is it better to modify the first one to include `us-east-1` than leaving out the `region` value?

>     2. They can use a different region by following the docs in this PR, but it is totally optional

Reply:
Maybe we can just remove the first example. It seems sufficient that the second example mentions the default is `us-east-1`, and if they want to change the region, this is how they do it.

Reply:
@travisn: Thanks. I have updated the PR accordingly